lib/src/multires_stochastic_texture_synthesis.rs

```diff
@@ -1191,6 +1201,213 @@ impl PrerenderedU8Function {
     }
 }
 
+struct TreeGrid {
+    grid_width: u32,
+    grid_height: u32,
+    offset_x: i32,
+    offset_y: i32,
+    chunk_size: u32,
+    rtrees: Vec<RwLock<RTree<[i32; 2]>>>,
+}
+
+// This is a grid of rtrees
+// The idea is that most pixels after the first couple steps will have their neighbors close by
+impl TreeGrid {
+    pub fn new(width: u32, height: u32, chunk_size: u32, offset_x: u32, offset_y: u32) -> TreeGrid {
+        let mut rtrees: Vec<RwLock<RTree<[i32; 2]>>> = Vec::new();
+        let grid_width = width / chunk_size + 1;
+        let grid_height = height / chunk_size + 1;
+        rtrees.resize_with((grid_width * grid_height) as usize, || {
+            RwLock::new(RTree::new())
+        });
```

Comment:
Could just use `let rtrees = vec![RwLock::new(RTree::new()); grid_width * grid_height];`

Reply:
👍

Reply:
Unfortunately got an error trying to do this saying that RwLock<Rtree> does not implement the copy constructor